src/lib.rs

```diff
@@ -234,7 +278,7 @@ impl_global_alias!(sha3_512, 512);
 impl Keccak {
     pub fn new(rate: usize, delim: u8) -> Keccak {
```

Comment:
Should probably panic early in case `rate == 0` to avoid overflows later on when we subtract 1